### PR TITLE
Feature: Add support for SHA256_PROCESS_BLOCK syscall

### PR DESCRIPTION
**Pull request by HermanObst**
Problem: We currently dont support SHA256_PROCESS_BLOCK, this PR implement it

## Breaking changes?

- [ ] yes
- [x] no

### CODE REVIEW

File: crates/starknet-os/src/execution/syscall_handler.rs [hunk 1]
```diff
@@ -671,3 +674,76 @@ where
...
+                .to_bigint()
+                .to_u32()
+                .expect("libfunc only accepts SHA256StateHandle which can only be created from an Array<u32>.")
+        });
+
+        sha2::compress256(&mut state_as_words, &[data_as_bytes]);
+
+        let segment = eh_ref.sha256_segment.unwrap_or(vm.add_memory_segment());
+
+        let response = (segment + 24)?; // 'out_state' is 24 bytes into 'struct Sha256ProcessBlock'
```
Comment (notlesh) [inline crates/starknet-os/src/execution/syscall_handler.rs:740]: I should have used `FieldOffsetGetters` here

File: crates/starknet-os/src/hints/execute_transactions.rs [hunk 2]
```diff
@@ -133,3 +134,36 @@ pub fn fill_holes_in_rc96_segment(
 
     Ok(())
 }
+
+#[allow(unused)]
```
Comment (odesenfans) [inline crates/starknet-os/src/hints/execute_transactions.rs:138]: Remove this.

File: crates/starknet-os/src/hints/execute_transactions.rs [hunk 3]
```diff
@@ -133,3 +134,36 @@ pub fn fill_holes_in_rc96_segment(
...
+pub const HINT_1: &str = indoc! {r#"# Add dummy pairs of input and output.
```
Comment (odesenfans) [inline crates/starknet-os/src/hints/execute_transactions.rs:139]: Use a better name. `SHA2_FINALIZE`?

File: crates/starknet-os/src/hints/execute_transactions.rs [hunk 4]
```diff
@@ -133,3 +134,36 @@ pub fn fill_holes_in_rc96_segment(
...
+_sha256_input_chunk_size_felts = ids.SHA256_INPUT_CHUNK_SIZE_FELTS
+assert 0 <= _sha256_input_chunk_size_felts < 100
+
+message = [0] * _sha256_input_chunk_size_felts
+w = compute_message_schedule(message)
+output = sha2_compress_function(IV, w)
+padding = (message + IV + output) * number_of_missing_blocks
+segments.write_arg(ids.sha256_ptr_end, padding)"#};
+
+pub fn hint_1(
```
Comment (odesenfans) [inline crates/starknet-os/src/hints/execute_transactions.rs:157]: Use a better name. `sha2_finalize` and rename the implementation to `sha2_finalize_impl`?

**Review by odesenfans — CHANGES_REQUESTED** [review]
Minor comments.

**Review by odesenfans — APPROVED** [review]

File: crates/starknet-os/src/hints/execute_transactions.rs (hunk 4 above)
Comment (notlesh) [reply crates/starknet-os/src/hints/execute_transactions.rs]: FYI, the names `hint_n` come from the [hint_tool](https://github.com/keep-starknet-strange/snos/blob/7cde76693b37e160e49c3b7cd5864a7951e682e4/crates/bin/hint_tool/src/main.rs#L142)

**Pull request merged**